### PR TITLE
ElementDeleterBase fixes for DistributedMesh

### PR DESCRIPTION
**Pull request by permcody**
Fixup ElementDeleterBase

### CODE REVIEW

**Comment by permcody** [comment]
Supersedes #8656

File: framework/src/meshmodifiers/ElementDeleterBase.C [hunk 1]
```diff
@@ -56,7 +56,28 @@ ElementDeleterBase::modify()
    *       but because the order of deletion might impact what happens to any existing sidesets or nodesets.
    */
   for (const auto & elem : deleteable_elems)
+  {
+    // On distributed meshes, we'll need neighbor links to be useable
+    // shortly, so we can't just leave dangling pointers.
+    //
+    // FIXME - this could be made AMR-aware and refactored into
```
Comment (aeslaughter) [inline framework/src/meshmodifiers/ElementDeleterBase.C]: Do you really want to put this FIXME comment, here. Shouldn't you just put on issue on libMesh to add this feature.

**Review by aeslaughter — DISMISSED** [review]
I have one rather minor comment to address, otherwise it looks fine.

File: framework/src/meshmodifiers/ElementDeleterBase.C (hunk 1 above)
Comment (permcody) [inline framework/src/meshmodifiers/ElementDeleterBase.C]: This is fine. It's not really an issue.

**Review by jwpeterson — APPROVED** [review]

**Pull request merged**